org.eclipse.richbeans.api/src/org/eclipse/richbeans/api/binding/IBeanController.java

```diff
@@ -61,6 +61,14 @@
 	 */
 	T getBean();
 	
+	/**
+	 * Calling this method assigns the linked bean to some other data
+	 * and it calls a beanToUI() to update the UI.
+	 * @param model
+	 * @throws Exception from the beanToUI() if the bean cannot update the UI
+	 */
+	T setBean(T model)  throws Exception;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)